src/main/java/io/aiven/kafka/connect/common/config/AivenCommonConfig.java

```diff
@@ -223,7 +223,9 @@ public final Template getFilenameTemplate() {
     protected final void validateKeyFilenameTemplate() {
         // Special checks for {{key}} filename template.
         final Template filenameTemplate = getFilenameTemplate();
-        if (RecordGrouperFactory.KEY_RECORD.equals(RecordGrouperFactory.resolveRecordGrouperType(filenameTemplate))) {
+        final String groupType = RecordGrouperFactory.resolveRecordGrouperType(filenameTemplate);
+        if (RecordGrouperFactory.KEY_RECORD.equals(groupType) 
+                || RecordGrouperFactory.TOPIC_PARTITION_KEY_RECORD.equals(groupType)) {
             if (getMaxRecordsPerFile() > 1) {
```

Comment:
maybe move this logic to the factory, within a method `RecordGrouperFactory.isKeyBased(groupType)`